From 01a517aee9ad501c2b7dd90e44e316ffb8e2c35d Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/runtime/pull/62938
From: Antoine Martin <dev@ayakael.net>
Date: Sun, 2 Jan 2022 22:00:09 +0000
Subject: [PATCH 2/2] add RID for Alpine 3.15

As of version 5.0.12, runtime does not have the RIDs for Alpine Linux 3.15.
This patch adds them.

---
 .../runtime.compatibility.json                | 114 ++++++++++++++++++
 .../Microsoft.NETCore.Platforms/runtime.json  |  23 ++++
 .../runtimeGroups.props                       |   2 +-
 3 files changed, 138 insertions(+), 1 deletion(-)

diff --git a/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json b/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
index 844590f8022..6ff9cdd2cae 100644
--- a/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
+++ b/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
@@ -508,6 +508,120 @@
     "any",
     "base"
   ],
+  "alpine.3.15": [
+    "alpine.3.15",
+    "alpine.3.14",
+    "alpine.3.13",
+    "alpine.3.12",
+    "alpine.3.11",
+    "alpine.3.10",
+    "alpine.3.9",
+    "alpine.3.8",
+    "alpine.3.7",
+    "alpine.3.6",
+    "alpine",
+    "linux-musl",
+    "linux",
+    "unix",
+    "any",
+    "base"
+  ],
+  "alpine.3.15-arm": [
+    "alpine.3.15-arm",
+    "alpine.3.15",
+    "alpine.3.14-arm",
+    "alpine.3.14",
+    "alpine.3.13-arm",
+    "alpine.3.13",
+    "alpine.3.12-arm",
+    "alpine.3.12",
+    "alpine.3.11-arm",
+    "alpine.3.11",
+    "alpine.3.10-arm",
+    "alpine.3.10",
+    "alpine.3.9-arm",
+    "alpine.3.9",
+    "alpine.3.8-arm",
+    "alpine.3.8",
+    "alpine.3.7-arm",
+    "alpine.3.7",
+    "alpine.3.6-arm",
+    "alpine.3.6",
+    "alpine-arm",
+    "alpine",
+    "linux-musl-arm",
+    "linux-musl",
+    "linux-arm",
+    "linux",
+    "unix-arm",
+    "unix",
+    "any",
+    "base"
+  ],
+  "alpine.3.15-arm64": [
+    "alpine.3.15-arm64",
+    "alpine.3.15",
+    "alpine.3.14-arm64",
+    "alpine.3.14",
+    "alpine.3.13-arm64",
+    "alpine.3.13",
+    "alpine.3.12-arm64",
+    "alpine.3.12",
+    "alpine.3.11-arm64",
+    "alpine.3.11",
+    "alpine.3.10-arm64",
+    "alpine.3.10",
+    "alpine.3.9-arm64",
+    "alpine.3.9",
+    "alpine.3.8-arm64",
+    "alpine.3.8",
+    "alpine.3.7-arm64",
+    "alpine.3.7",
+    "alpine.3.6-arm64",
+    "alpine.3.6",
+    "alpine-arm64",
+    "alpine",
+    "linux-musl-arm64",
+    "linux-musl",
+    "linux-arm64",
+    "linux",
+    "unix-arm64",
+    "unix",
+    "any",
+    "base"
+  ],
+  "alpine.3.15-x64": [
+    "alpine.3.15-x64",
+    "alpine.3.15",
+    "alpine.3.14-x64",
+    "alpine.3.14",
+    "alpine.3.13-x64",
+    "alpine.3.13",
+    "alpine.3.12-x64",
+    "alpine.3.12",
+    "alpine.3.11-x64",
+    "alpine.3.11",
+    "alpine.3.10-x64",
+    "alpine.3.10",
+    "alpine.3.9-x64",
+    "alpine.3.9",
+    "alpine.3.8-x64",
+    "alpine.3.8",
+    "alpine.3.7-x64",
+    "alpine.3.7",
+    "alpine.3.6-x64",
+    "alpine.3.6",
+    "alpine-x64",
+    "alpine",
+    "linux-musl-x64",
+    "linux-musl",
+    "linux-x64",
+    "linux",
+    "unix-x64",
+    "unix",
+    "any",
+    "base"
+  ],
   "alpine.3.6": [
     "alpine.3.6",
     "alpine",
diff --git a/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.json b/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.json
index 7d6aa102b50..a663a88166e 100644
--- a/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.json
+++ b/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.json
@@ -138,6 +138,29 @@
         "alpine.3.13-x64"
       ]
     },
+    "alpine.3.15": {
+      "#import": [
+        "alpine.3.14"
+      ]
+    },
+    "alpine.3.15-arm": {
+      "#import": [
+        "alpine.3.15",
+        "alpine.3.14-arm"
+      ]
+    },
+    "alpine.3.15-arm64": {
+      "#import": [
+        "alpine.3.15",
+        "alpine.3.14-arm64"
+      ]
+    },
+    "alpine.3.15-x64": {
+      "#import": [
+        "alpine.3.15",
+        "alpine.3.14-x64"
+      ]
+    },
     "alpine.3.6": {
       "#import": [
         "alpine"
diff --git a/src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props b/src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
index 3cb3d2df858..c7a346d1672 100644
--- a/src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
+++ b/src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
@@ -16,7 +16,7 @@
     <RuntimeGroup Include="alpine">
       <Parent>linux-musl</Parent>
       <Architectures>x64;arm;arm64</Architectures>
-      <Versions>3.6;3.7;3.8;3.9;3.10;3.11;3.12;3.13;3.14</Versions>
+      <Versions>3.6;3.7;3.8;3.9;3.10;3.11;3.12;3.13;3.14;3.15</Versions>
     </RuntimeGroup>
 
     <RuntimeGroup Include="android">
-- 
2.34.1

